### DISCUSSION TITLE
Can the real scene content be partially blocked?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#1400] [Q&A] [answered]
Hi, When displaying the real scene, I want to use the mouse to drag a rectangular box, and then the real scene in the rectangular box will not be displayed.
I want to implement such a function, what are some good suggestions? Thank you.

**Comment by pmconne** [comment, marked as answer]
When your question is "how can I do X", a good way to find an answer is to browse the sample showcase for a sample that does something similar and look up the docs for the APIs used by that sample.

You are asking to [clip the view](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=view-clip-sample&imodel=Retail+Building+Sample). You can probably trivially modify or extend the [ViewClipByShapeTool](https://www.itwinjs.org/reference/imodeljs-frontend/tools/viewclipbyshapetool/).

If you need an example of how to use mouse drag to define and draw a rectangle, [here's one](https://github.com/imodeljs/imodeljs/blob/8fd9f3fa67bd159d4323ca4ce622b33427a2de29/core/frontend/src/tools/ViewTool.ts#L3554).

**Reply by shenshengyi** [reply]
ok, thank you very much for your answers.
I will learn the usage of ViewClipByShapeTool.
